### ISSUE TITLE
Failure to configure with python bindings on Ubuntu

### ISSUE THREAD

**Issue by agrayver** [dealii/dealii#8961]
I get this error when trying to compile with python bindings:

> -- Setting up python bindings
> -- Found PythonInterp: /home/ag/anaconda3/bin/python (found version "3.7.3") 
> -- Found PythonLibs: /home/ag/anaconda3/lib/libpython3.7m.so (found version "3.7.3") 
> CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:801 (message):
>   New Boost version may have incorrect or missing dependencies and imported
>   targets
> Call Stack (most recent call first):
>   /usr/share/cmake-3.10/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
>   /usr/share/cmake-3.10/Modules/FindBoost.cmake:1558 (_Boost_MISSING_DEPENDENCIES)
>   contrib/python-bindings/source/CMakeLists.txt:38 (_FIND_PACKAGE)
>
> -- Could NOT find Boost
> CMake Error at contrib/python-bindings/source/CMakeLists.txt:42 (MESSAGE):
>   DEAL_II_COMPONENT_PYTHON_BINDINGS has unmet configuration requirements: The
>   external boost library does not provide Boost.Python

This error comes from this snippet in '/dealii/contrib/python-bindings/source':

```
#
# Unset Boost_Found and run the low level FindBOOST.cmake module again to
# pick up libboost_python.so
#
LIST(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
SET(Boost_FOUND)
# Use the low level _FIND_PACKAGE function instead of our wrapper
_FIND_PACKAGE(Boost 1.54 COMPONENTS python)
LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)

IF(NOT Boost_FOUND)
  MESSAGE(FATAL_ERROR
    "DEAL_II_COMPONENT_PYTHON_BINDINGS has unmet configuration requirements: "
    "The external boost library does not provide Boost.Python"
    )
ENDIF()
```
This indeed seems fragile... The external boost.python v. 1.71 is compiled and installed correctly and passes independent tests. Is there a better way to add boost.python to the configuration?

**Comment by agrayver** [comment]
I tried with a new CMake, but it still does not add the suffix itself, although it [documents](https://cmake.org/cmake/help/v3.11/module/FindBoost.html) this as of 3.11:

> Note that Boost Python components require a Python version suffix (Boost 1.67 and later), e.g. python36 or python27 for the versions built against Python 3.6 and 2.7, respectively. This also applies to additional components using Python including mpi_python and numpy. Earlier Boost releases may use distribution-specific suffixes such as 2, 3 or 2.7. These may also be used as suffixes, but note that they are not portable.

From boost 1.67 on, one has to do it manually:

`_FIND_PACKAGE(Boost 1.67 COMPONENTS python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR} REQUIRED)`

Shall we fix this in deal.II by adding the above line and bumping the minimum version (or conditionally turning back to the present way for < boost 1.67)?

**Comment by masterleinad** [comment]
> Shall we fix this in deal.II by adding the above line and bumping the minimum version (or conditionally turning back to the present way for < boost 1.67)?

If you could provide a pull request just for making it work with boost version >= 1.67 that would be great. Let's not bump the required version, yet (although we are requiring boost 1.59 anyway, so we could just go with that version still being backward-compatible).

**Comment by agrayver** [comment]
Resolved by #8968
